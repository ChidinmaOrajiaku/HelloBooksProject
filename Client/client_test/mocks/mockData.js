import { updatePassword } from "../../actions/updatePassword";

const mockData = {
  userDetails: {
    message: 'Welcome admin',
    username: 'admin',
    token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImRhZGR5Iiwicm9sZSI6InVzZXIiLCJpZCI6MiwiaWF0IjoxNTEzODc1ODE3LCJleHAiOjE1MTM5NjIyMTd9.oexUFhi08vhMU88LmZJvRDy9JFdThMpfb3Qz9DWWzgk'
  },
  userDetailsResponse: {
    username: 'daddy',
    role: 'user',
    id: 2,
    iat: 1513875817,
    exp: 1513962217
  },
  userCountResponse: {
    count: 7
  },
  borrowBookData: {
    bookId: 1,
    userId: 1,
  },
  borrowBookDataResponse: {
    returned: false,
    id: 24,
    userId: 1,
    bookId: 7,
    toReturnDate: '2018-01-14',
    updatedAt: '2017-12-21T21:16:08.442Z',
    createdAt: '2017-12-21T21:16:08.442Z',
    returnDate: null,
    categoryId: null
  },
  createBookData: {
    title: 'a',
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorData: {
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorResponse: 'An error occured',
  createBookResponse: {
    message: 'Succesfully added'
  },
  getABookData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBorrowedBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getCategoryData: {
    category: 'Fiction'
  },
  getUserData: {
    firstname: 'daddy',
    lastname: 'daddy',
    username: 'daddy',
    password: 'daddy',
    email: 'daddy@gmail.com'
  },
  bookData: {
    title: 'a',
    author: 'b',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  returnBookData: {
    message: 'Successfully Returned'
  },
  userData: {
    verifyPassword: '1234',
    password: '4321'
  },
  errorMessage: 'An error occurred',
  image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1515403427/ena7dhy6xriztom6vuvo.jpg'
};

export default mockData;
