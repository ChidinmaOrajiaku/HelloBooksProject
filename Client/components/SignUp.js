import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import BodyNavigationBar from './BodyNavigationBar';
import Footer from './Footer';
import map from 'lodash/map';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { userSignupRequest } from '../actions/signupAction';

/**
 * 
 * 
 * @class SignUp
 * @extends {React.Component}
 */
class SignUp extends React.Component {
  /**
   * Creates an instance of SignUp.
   * @param {any} props 
   * @memberof SignUp
   */
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      username: '',
      password: '',
      email: '',
      errors: false,
      success: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.onSignupSubmit = this.onSignupSubmit.bind(this);
  }
  /**
 * 
 * 
 * @param {any} event 
 * @memberof SignUp
 */
  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  /**
   * 
   * 
   * @param {any} event 
   * @memberof SignUp
   */
  onSignupSubmit(event) {
    event.preventDefault();
    this.props.userSignupRequest(this.state).then(
      (success) => {
        Materialize.toast(success.data.message, 2000, 'teal rounded');
        this.setState({ success: success.data.message });
      },
      (errors) => {
        Materialize.toast(errors.response.data, 2000, 'red accent-3 rounded');
        this.setState({ errors: errors.response.data });
      }
    );
  }

  /**
   * 
   * 
   * @returns 
   * @memberof SignUp
   */
  render() {
    const { userSignupRequest } = this.props;
    const { errors, success } = this.state;
    return (
      <div className="signUp">
        <div className="bodyNavigation"> <BodyNavigationBar /> </div>
        <div className="row container">
          <div className="quotes">
            <h1 className="books">“The person, be it gentleman or lady, who has not pleasure in a good novel, must be intolerably stupid.”</h1>
            <h2 className="author"> ~Jane Austen, Northanger Abbey</h2>
          </div>
          <div className="col m6 offset-m3">
            <div className="card">
              <div className="card-content black-text">
                <form onSubmit={this.onSignupSubmit} id="form">
                  <div className="row">
                    <div className="input-field col s6">
                      <input value={this.state.firstname} onChange={this.handleChange} id="firstname" required="required" type="text" className="validate"/>
                      <label htmlFor="firstname">First Name</label>
                    </div>
                    <div className="input-field col s6">
                      <input value={this.state.lastname} onChange={this.handleChange} id="lastname" required="required" type="text" className="validate"/>
                      <label htmlFor="lastname">Last Name</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input value={this.state.username} onChange={this.handleChange} id="username" required="required" type="text" className="validate"/>
                      <label htmlFor="username">User Name</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input value={this.state.password} onChange={this.handleChange} id="password" required="required" type="password" className="validate"/>
                      <label htmlFor="password">Password</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input value={this.state.email} onChange={this.handleChange} id="email" required="required" type="email" className="validate"/>
                      <label htmlFor="email">Email</label>
                    </div>
                  </div>
                  <button className="btn waves-effect waves-light" type="submit" name="action">Sign Up
                    <i className="material-icons right">send</i>
                  </button>
                </form>
              </div>
              <div className="account">
                <p> I already have an account? <Link to="/login"> Login </Link> </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SignUp.propTypes = {
  userSignupRequest: PropTypes.func.isRequired
};

export default connect(null, { userSignupRequest })(SignUp);
