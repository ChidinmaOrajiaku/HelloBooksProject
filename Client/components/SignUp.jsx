import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import BodyNavigationBar from './BodyNavigationBar';
import Footer from './Footer';
import { userSignupRequest } from '../actions/signupAction';

/**
 *
 *
 * @class SignUp
 * @extends {React.Component}
 */
export class SignUp extends React.Component {
  /**
   * Creates an instance of SignUp.
   * @constructor
   * @param {any} props
   * @memberof SignUp
   */
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      username: '',
      password: '',
      email: '',
      errors: false,
      success: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.onSignupSubmit = this.onSignupSubmit.bind(this);
  }

  /**
 * @returns {object} response object
 * @param {any} event
 * @memberof SignUp
 */
  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  /**
   * Handles sign up action
   * @returns {object} response object
   * @param {any} event
   * @memberof SignUp
   */
  onSignupSubmit(event) {
    event.preventDefault();
    this.props.userSignupRequest(this.state).then(() => {
      setTimeout(() => {
        if (this.props.usersId.isAuthenticated === true) {
          this.props.history.push('/profile');
        } else if (this.props.signUpResponse.isSignedUp === false) {
          Materialize.toast(this.props.signUpResponse.error.response.data, 2000, 'red rounded');
        }
      }, 1000);
    });
  }

  /**
   * React Element Markup
   * @returns {object} response object
   * @memberof SignUp
   */
  render() {
    return (
      <div className="signUp">
        <div className="bodyNavigation"> <BodyNavigationBar /> </div>
        <div className="row container">
          <div className="col s12 m6 offset-m3">
            <div className="card" data-aos="flip-left">
              <div className="card-content black-text">
                <form
                  onSubmit={this.onSignupSubmit}
                  id="form">
                  <div className="row">
                    <div className="input-field col s6">
                      <input
                        value={this.state.firstname}
                        onChange={this.handleChange}
                        id="firstname"
                        required="required"
                        type="text"
                        className="validate"
                      />
                      <label htmlFor="firstname">First Name</label>
                    </div>
                    <div className="input-field col s6">
                      <input
                        value={this.state.lastname}
                        onChange={this.handleChange}
                        id="lastname"
                        required="required"
                        type="text"
                        className="validate"
                      />
                      <label htmlFor="lastname">Last Name</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input
                        value={this.state.username}
                        onChange={this.handleChange}
                        id="username"
                        required="required"
                        type="text"
                        className="validate"
                      />
                      <label htmlFor="username">User Name</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input
                        value={this.state.password}
                        onChange={this.handleChange}
                        id="password"
                        required="required"
                        type="password"
                        className="validate"
                      />
                      <label htmlFor="password">Password</label>
                    </div>
                  </div>
                  <div className="row">
                    <div className="input-field col s12">
                      <input
                        value={this.state.email}
                        onChange={this.handleChange}
                        id="email"
                        required="required"
                        type="email"
                        className="validate"/>
                      <label htmlFor="email">Email</label>
                    </div>
                  </div>
                  <button
                    className="btn waves-effect waves-light"
                    type="submit"
                    name="action">
                    Sign Up
                    <i className="material-icons right">send</i>
                  </button>
                </form>
              </div>
              <div className="account">
                <p> I already have an account?
                  <Link to="/login"> Login
                  </Link>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SignUp.propTypes = {
  userSignupRequest: PropTypes.func.isRequired
};

export const mapStateToProps = state => (
  {
    usersId: state.auth,
    signUpResponse: state.signUp[0],
  }
);

export default connect(mapStateToProps, { userSignupRequest })(SignUp);
