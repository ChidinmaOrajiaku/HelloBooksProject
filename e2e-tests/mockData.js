module.exports = {
  file: {
    lastModified: 1508779018000,
    lastModifiedDate: 'Mon Oct 23 2017 18:16:58 GMT+0100 (WAT)',
    name: '3 card game.jpg',
    size: 642971,
    type: 'image/jpeg',
    webkitRelativePath: ''
  },
};

