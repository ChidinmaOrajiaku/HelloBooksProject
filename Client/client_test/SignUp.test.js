import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockData from './mocks/mockData';
import { SignUp } from '../components/SignUp.jsx';

configure({ adapter: new Adapter() });

describe('<SignUp />', () => {
  const props = {
    userSignupRequest: jest.fn(() => Promise.resolve())
  };

  let mountedComponent;
  const signupFormItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<SignUp {...props}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('renders <SignUp /> component', () => {
    expect(signupFormItem()).toHaveLength(1);
  });

  it('should call handleChange method', () => {
    const spy = jest.spyOn(signupFormItem().instance(), 'handleChange');
    const event = {
      preventDefault: jest.fn(),
      target: {
        id: 'review',
        value: 'chideberecom'
      }
    };
    signupFormItem().instance().handleChange(event);
    expect(spy).toHaveBeenCalled();
  });

  it('should call onSignupSubmit method', () => {
    const { getSignUpData } = mockData;
    const spy = jest.spyOn(signupFormItem().instance(), 'onSignupSubmit');
    const event = {
      preventDefault: jest.fn(),
    };
    signupFormItem().instance().onSignupSubmit(event);
    expect(spy).toHaveBeenCalled();
    const form = signupFormItem().find('form');
    signupFormItem().setState(getSignUpData);
    form.simulate('submit', event);
  });

  it('should update state on first name field change', () => {
    signupFormItem().find('#firstname').simulate('change', {
      target: {
        id: 'firstname',
        value: 'chidebere'
      },
    });
    expect(signupFormItem().state().firstname).toBe('chidebere');
  });

  it('should update state on last name field change', () => {
    signupFormItem().find('#lastname').simulate('change', {
      target: {
        id: 'lastname',
        value: 'chidebere'
      },
    });
    expect(signupFormItem().state().lastname).toBe('chidebere');
  });

  it('should update state on username field change', () => {
    signupFormItem().find('#username').simulate('change', {
      target: {
        id: 'username',
        value: 'chidebere'
      },
    });
    expect(signupFormItem().state().username).toBe('chidebere');
  });

  it('should update state on email field change', () => {
    signupFormItem().find('#email').simulate('change', {
      target: {
        id: 'email',
        value: 'chidebere@w.com'
      },
    });
    expect(signupFormItem().state().email).toBe('chidebere@w.com');
  });

  it('should update state on password field change', () => {
    signupFormItem().find('#password').simulate('change', {
      target: {
        id: 'password',
        value: 'password'
      },
    });
    expect(signupFormItem().state().password).toBe('password');
  });

  it('should match snapshot test', () => {
    const component = signupFormItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});

