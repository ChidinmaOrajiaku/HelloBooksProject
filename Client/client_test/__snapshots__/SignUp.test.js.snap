// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot test 1`] = `
<div
  className="signUp"
>
  <div
    className="bodyNavigation"
  >
     
    <Component />
     
  </div>
  <div
    className="row container"
  >
    <div
      className="col s12 m6 offset-m3"
    >
      <div
        className="card"
        data-aos="flip-left"
      >
        <div
          className="card-content black-text"
        >
          <form
            id="form"
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s6"
              >
                <input
                  className="validate"
                  id="firstname"
                  onChange={[Function]}
                  required="required"
                  type="text"
                  value=""
                />
                <label
                  htmlFor="firstname"
                >
                  First Name
                </label>
              </div>
              <div
                className="input-field col s6"
              >
                <input
                  className="validate"
                  id="lastname"
                  onChange={[Function]}
                  required="required"
                  value=""
                  ype="text"
                />
                <label
                  htmlFor="lastname"
                >
                  Last Name
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="username"
                  onChange={[Function]}
                  required="required"
                  type="text"
                  value=""
                />
                <label
                  htmlFor="username"
                >
                  User Name
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="password"
                  onChange={[Function]}
                  required="required"
                  type="password"
                  value=""
                />
                <label
                  htmlFor="password"
                >
                  Password
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  className="validate"
                  id="email"
                  onChange={[Function]}
                  required="required"
                  type="email"
                  value=""
                />
                <label
                  htmlFor="email"
                >
                  Email
                </label>
              </div>
            </div>
            <button
              className="btn waves-effect waves-light"
              name="action"
              type="submit"
            >
              Sign Up
              <i
                className="material-icons right"
              >
                send
              </i>
            </button>
          </form>
        </div>
        <div
          className="account"
        >
          <p>
             I already have an account?
            <Link
              replace={false}
              to="/login"
            >
               Login
            </Link>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
